cf/rds-agent.yaml

```diff
@@ -0,0 +1,119 @@
+    Type: String
+    Default: aurora-staging
+
+Resources:
+  SSMProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties: 
+      InstanceProfileName: SSMInstanceProfile
+      Roles: 
+        - !Ref SSMRole
+
+  SSMRole:
+    Type: AWS::IAM::Role
+    Properties: 
+      RoleName: SSMInstanceProfile
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Description: Basic SSM permissions for EC2
+      ManagedPolicyArns: 
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+      Policies:
+        - PolicyName: "can-access-rds"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "rds-db:connect"
+                Resource: 
+                  # - arn:aws:rds-db:eu-west-1:*:dbuser:*/* 
+                  - !Sub "arn:aws:rds-db:eu-west-1:${AWS::AccountId}:dbuser:*/dev_role"
+
+  RDSAgent:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: t2.micro
+      IamInstanceProfile: !Ref SSMProfile
+      ImageId: !Ref LatestAmiId
+      SecurityGroupIds:
+        - Fn::ImportValue: !Sub "biomage-${Environment}-rds::RDSAgentSecurityGroupId"
+      SubnetId:
+      #Pick first element in SubnetsPrivate
+        !Select [ 0, Fn::Split: [',', Fn::ImportValue: !Sub "eksctl-biomage-${Environment}-cluster::SubnetsPrivate"] ]
+
+      #UserData Runs when ec2 is being deployed
+      UserData:
+        Fn::Base64:
+          Fn::Sub:
+            - |
+              Content-Type: multipart/mixed; boundary="//"
+              MIME-Version: 1.0
+
+              --//
+              Content-Type: text/cloud-config; charset="us-ascii"
+              MIME-Version: 1.0
+              Content-Transfer-Encoding: 7bit
+              Content-Disposition: attachment; filename="cloud-config.txt"
+
+              #cloud-config
+              cloud_final_modules:
+              - [scripts-user, always]
+
+              --//
+              Content-Type: text/x-shellscript; charset="us-ascii"
+              MIME-Version: 1.0
+              Content-Transfer-Encoding: 7bit
+              Content-Disposition: attachment; filename="userdata.txt"
+
+              #!/bin/bash -e
+              sudo yum -y install postgresql
+              PGPASSWORD='${DBPassword}' psql \
+                --host=aurora-cluster-staging.cluster-cuhf24s6layy.eu-west-1.rds.amazonaws.com \
+                --port=5432 \
+                --username=${DBUsername} \
+                --dbname=auroraDB \
+                -c 'CREATE ROLE api_role WITH LOGIN;
+                  CREATE ROLE dev_role WITH LOGIN;
+                  GRANT rds_iam TO api_role;
+                  GRANT rds_iam, ${DBUsername} TO dev_role;
```

Comment:
Granting admin to dev_role, we could also choose to do something less involved here maybe.